### PR TITLE
Adding high level arch description

### PR DESCRIPTION
**Pull request by krish7919**

### CODE REVIEW

File: docs/server/source/production-deployment-template/architecture.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+============================
+
+An IPDB Production deployment is hosted on a Kubernetes cluster and includes:
+
+* NGINX, OpenResty, BigchainDB and MongoDB
+  `Kubernetes Services <https://kubernetes.io/docs/concepts/services-networking/service/>`_.
+* NGINX, OpenResty, BigchainDB, Monitoring Agent and Backup Agent
+  `Kubernetes Deployments <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>`_.
+* MongoDB `Kubernetes StatefulSet <https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/>`_.
+* External systems like `3scale <https://3scale.net>`_,
```
Comment (muawiakh) [inline docs/server/source/production-deployment-template/architecture.rst]: How about 3rd party services/infrastrucutre? Also we can remove the `like`, since we only have these.

Comment (krish7919) [reply docs/server/source/production-deployment-template/architecture.rst]: We also use a certificate signing authorities and DNS providers. But these are the main ones.
Reword it to `Third party systems like...`?

File: docs/server/source/_static/arch.svg [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
```
Comment (muawiakh) [inline docs/server/source/_static/arch.svg]: It is just a visual/graphic thing, but can we shift the components slightly to the right-> towards the middle. Currently they are more aligned on the right. Only do it if it is a quick thing.
screen shot 2017-09-26 at 11 09 14 am

File: docs/server/source/production-deployment-template/architecture.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  `MongoDB Cloud Manager <https://cloud.mongodb.com>`_ and the
+  `Azure Operations Management Suite
+  <https://docs.microsoft.com/en-us/azure/operations-management-suite/>`_.
+
+.. image:: ../_static/arch.svg
+
+
+We describe the role of each of these modules here.
```
Comment (muawiakh) [inline docs/server/source/production-deployment-template/architecture.rst]: nit: Can we remove this line or rephrase? Or a subheading `Modules` or `Roles` that has sub-headings for each component.

File: docs/server/source/production-deployment-template/architecture.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+   to BigchainDB and terminates at NGINX for now.
+
+#. Request Routing: For HTTPS connections on port 443 (or the configured BigchainDB public api port),
+   the connection is proxied to:
+
+   #. OpenResty Service if it is a POST request.
+   #. BigchainDB Service if it is a GET request.
+
+
+We use an NGINX TCP proxy on port 27017 at the cloud entrypoint for:
```
Comment (shahbazn) [inline docs/server/source/production-deployment-template/architecture.rst]: do we need to expose specific ports here?

File: docs/server/source/production-deployment-template/architecture.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  `Kubernetes Services <https://kubernetes.io/docs/concepts/services-networking/service/>`_.
+* NGINX, OpenResty, BigchainDB, Monitoring Agent and Backup Agent
+  `Kubernetes Deployments <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>`_.
+* MongoDB `Kubernetes StatefulSet <https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/>`_.
+* External systems like `3scale <https://3scale.net>`_,
+  `MongoDB Cloud Manager <https://cloud.mongodb.com>`_ and the
+  `Azure Operations Management Suite
+  <https://docs.microsoft.com/en-us/azure/operations-management-suite/>`_.
+
+.. image:: ../_static/arch.svg
```
Comment (shahbazn) [inline docs/server/source/production-deployment-template/architecture.rst]: when I try to view this most of the text shows not supported by viewer

File: docs/server/source/_static/arch.svg [hunk 6]
```diff
@@ -0,0 +1,2 @@
...
...t-size: 12px; font-family: Helvetica; color: rgb(0, 0, 0); line-height: 1.2; vertical-align: top; white-space: nowrap;"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;text-align:inherit;text-decoration:inherit;">MongoDB Cluster<div>Communication</div></div></div></foreignObject><text x="48" y="19" fill="#000000" text-anchor="middle" font-size="12px" font-family="Helvetica">MongoDB Cluster&lt;div&gt;Communication&lt;/div&gt;</text></switch></g><g transform="translate(254.5,719.5)scale(0.8)"><switch><foreignObject style="overflow:visible;" pointer-events="all" width="62" height="12" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; font-size: 12px; font-family: Helvetica; color: rgb(0, 0, 0); line-height: 1.2; vertical-align: top; white-space: nowrap; font-weight: bold;"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;text-align:inherit;text-decoration:inherit;">IPDB Node</div></div></foreignObject><text x="31" y="12" fill="#000000" text-anchor="middle" font-size="12px" font-family="Helvetica" font-weight="bold">IPDB Node</text></switch></g></g></svg>
```
Comment (shahbazn) [inline docs/server/source/_static/arch.svg]: nit pick: may be we don't need to show a distinction b/w service and deployment for simplicity of the logic diagram.

File: docs/server/source/production-deployment-template/architecture.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
+Architecture of an IPDB Node
+============================
+
+An IPDB Production deployment is hosted on a Kubernetes cluster and includes:
+
+* NGINX, OpenResty, BigchainDB and MongoDB
+  `Kubernetes Services <https://kubernetes.io/docs/concepts/services-networking/service/>`_.
+* NGINX, OpenResty, BigchainDB, Monitoring Agent and Backup Agent
+  `Kubernetes Deployments <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>`_.
```
Comment (shahbazn) [inline docs/server/source/production-deployment-template/architecture.rst]: is mongoldb deployment just a proxy service since IPDB is using hosted mongoldb cloud?

Comment (krish7919) [reply docs/server/source/production-deployment-template/architecture.rst]: We do not use hosted MongoDB cloud for running MongoDB.
We only leverage MongoDB Cloud Manager for backups.

File: docs/server/source/_static/arch.svg (hunk 2 above)
Comment (krish7919) [reply docs/server/source/_static/arch.svg]: Let me see how to do that. Should be do-able.

File: docs/server/source/_static/arch.svg (hunk 6 above)
Comment (krish7919) [reply docs/server/source/_static/arch.svg]: We wanted that in this diagram. The earlier one [here](https://medium.com/ipdb-blog/using-the-ipdb-test-network-a615f93d50a9) did not have the separation.

File: docs/server/source/production-deployment-template/architecture.rst (hunk 3 above)
Comment (krish7919) [reply docs/server/source/production-deployment-template/architecture.rst]: Removed.

File: docs/server/source/production-deployment-template/architecture.rst (hunk 4 above)
Comment (krish7919) [reply docs/server/source/production-deployment-template/architecture.rst]: These are public ports and are always exposed.
I have added that these are configurable.

**Comment by ttmc** [comment]
Please add comments to note what the arrows mean (and don't mean). They don't mean the direction of communications (which is always bi-directional). A --> B means A connects to B? What does that mean for the order in which we deploy the things?

File: docs/server/source/production-deployment-template/architecture.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
+Architecture of an IPDB Node
```
Comment (shahbazn) [inline docs/server/source/production-deployment-template/architecture.rst]: nit: may be change the name to IPDB reference architecture just IPDB architecture? what does the node refer to here?

File: docs/server/source/production-deployment-template/architecture.rst [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+============================
+
+An IPDB Production deployment is hosted on a Kubernetes cluster and includes:
+
+* NGINX, OpenResty, BigchainDB and MongoDB
+  `Kubernetes Services <https://kubernetes.io/docs/concepts/services-networking/service/>`_.
+* NGINX, OpenResty, BigchainDB, Monitoring Agent and Backup Agent
+  `Kubernetes Deployments <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>`_.
+* MongoDB `Kubernetes StatefulSet <https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/>`_.
+* Third party systems like `3scale <https://3scale.net>`_,
```
Comment (shahbazn) [inline docs/server/source/production-deployment-template/architecture.rst]: Can we use third party services instead of third party systems here?

File: docs/server/source/production-deployment-template/architecture.rst (hunk 8 above)
Comment (krish7919) [reply docs/server/source/production-deployment-template/architecture.rst]: `Node` refers to a `BigchainDB Node in a Federation`.
I would suggest that the title remains the same as the entire docs are in the `production deployment template` and can be customized by users in whatever way they seems to be fit.

**Comment by krish7919** [comment]
> What does that mean for the order in which we deploy the things?

The order of deployment is independent of this diagram, as long as all the components are deployed, it will work.
